generators/base/generator-base.mjs

```diff
@@ -376,12 +348,7 @@ export default class JHipsterBaseGenerator extends PrivateBase {
    * @param {string} clientFramework - The name of the client framework
    */
   addIcon(iconName, clientFramework) {
```

Comment:
My idea for needle has changed to use the source object.
We can inject the needle like here:
https://github.com/jhipster/generator-jhipster/blob/c6e0ea8035cb1fb2a9343d8067f73c6589e7622b/generators/docker/generator.mjs#L47-L74

and use like here:
https://github.com/jhipster/generator-jhipster/blob/c6e0ea8035cb1fb2a9343d8067f73c6589e7622b/generators/docker/generator.mjs#L100-L118

source is shared, so once injected every generator can use it.

Reply:
For now I would move the api to the client generator.

Reply:
Awesome, fully agree that this is the way to go.
Another idea out of my mind would be to be able to detect if the target code has been customized by the user pre merge. we could for example compare the fragment code that is in the 'jhipster_upgrade' branch with the content of the current branch, see if there's a diff.
If there's diff: don't merge but update the jhipster_upgrade branch with new code.
If there isn't: merge and update the baseline branch.

It would lead to a more frictionless custom + generated code cohabitation and a better dev XP in the end.

Nevertheless, the purpose of the merge request I currently provide is not to modify the behavior of the current codebase, but just to extract the logic in order to make it more simple to do it in a second round. I always prefer step-by-step approach rather than big bang changes (and I'm not enough confident doing so)

Reply:
Moved the api to the client generator